src/include/misc.i

```diff
@@ -38,17 +54,16 @@ __wt_rdtsc(WT_SESSION_IMPL *session) {
 #if (defined __i386)
 	uint64_t x;
 
-	WT_UNUSED(session);
-
 	__asm__ volatile ("rdtsc" : "=A" (x));
+	__tsc_check_monotonic(session, &x);
 	return (x);
 #elif (defined __amd64)
-	uint64_t a, d;
-
-	WT_UNUSED(session);
+	uint64_t a, d, x;
 
 	__asm__ volatile ("rdtsc" : "=a" (a), "=d" (d));
-	return ((d << 32) | a);
+	x = (d << 32) | a;
+	__tsc_check_monotonic(session, &x);
+	return (x);
```

Comment:
We're not checking the `__wt_optrack_get_expensive_timestamp()` path in `__wt_rdtsc()` for time-travel, is that not possible?

If time travel is possible, I'd suggest changing `__wt_optrack_get_expensive_timestamp()` to call `__wt_epoch()` instead of `__wt_epoch_raw()`.